source/_posts/Planetscape-a-dystopian-escape-game-for-36C3.md

```diff
@@ -71,17 +71,40 @@ In order to orchestrate all different components of the game, Social Dist0rtion
 [THC][0] played a crucial role in the creation of the game, allowing us to meet our high standards in terms of development experience. But THC was more than that. Fast prototyping and automatic deploying gave us the piece of mind we needed during our last hours of work.
 
 ### Zero Knowledge Proof
-As mentioned earlier, every time a correct password is found, a transaction is sent to the smart contract. Ethereum transactions are public and we don't want players to sniff passwords submitted by others. To avoid that, we implemented a cheap zero-knowledge proof scheme. Roughly speaking, the scheme is the following:
-
-1. Given a valid `password`, a player's `player_address`:
-2. The dApp generates `seed = keccak256(password)`.
-3. The dApp generates `chapter_private_key` from `seed`.
-4. The dApp generates `signature = sign(address, chapter_private_key)`
-5. The dApp calls the `submit` method of the smart sontract passing `signature`.
-6. The smart contract checks the current `chapter` for `player_address` in its storage.
-7. The smart contract loads the `chapter_address` for `chapter`.
-8. The smart contracts calculates `signing_address` by doing an `ecrecover(player_address, signature)`.
-9. If `signing_address == chapter_address`, then the smart contract updates the current chapter for the player, moving them to the next chapter.
+As mentioned earlier, every time a correct password is found, a transaction is sent to the smart contract. Ethereum transactions are public and we don't want players to sniff passwords submitted by others. To avoid that, we implemented a cheap zero-knowledge proof scheme.
```

Comment:
Is "cheap" something zero-knowledge proof inventors say? What does it mean? I'm curious

Reply:
Cheap because it doesn't require expensive computation and it's quite easy to understand.